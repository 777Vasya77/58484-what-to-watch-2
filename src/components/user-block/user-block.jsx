import React from 'react';
import PropTypes from "prop-types";
import {Link} from 'react-router-dom';
import {selectors} from "~/selectors/selectors";
import {connect} from "react-redux";

const UserBlock = (props) => {
  const {isAuth, authUser} = props;
  return (
    <div className="user-block">
      {isAuth
        ? <div className="user-block__avatar">
          <Link to="/mylist">
            <img src={authUser.fullAvatarUrl} alt={authUser.name} width="63" height="63"/>
          </Link>
        </div>
        : <Link to='/login' style={{color: `unset`}}>Sign In</Link>}
    </div>
  );
};

export {UserBlock};

UserBlock.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  authUser: PropTypes.object,
};

const mapStateToProps = (state) => ({
  isAuth: selectors.user.isAuth(state),
  authUser: selectors.user.getAuthUser(state),
});

export default connect(mapStateToProps)(UserBlock);
