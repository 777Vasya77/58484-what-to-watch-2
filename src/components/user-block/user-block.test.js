import React from 'react';
import renderer from 'react-test-renderer';
import {UserBlock} from "~/components/user-block/user-block";

jest.mock(`react-router-dom`, () => ({
  Link: `Link`
}));

describe(`UserBlock component tests`, () => {
  it(`Component render correctly`, () => {
    const tree = renderer
        .create(
            <UserBlock isAuth={true} authUser={({})}/>
        ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
