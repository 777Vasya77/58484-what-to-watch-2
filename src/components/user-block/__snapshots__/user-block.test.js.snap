// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component tests Component render correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <Link
      to="/mylist"
    >
      <img
        height="63"
        width="63"
      />
    </Link>
  </div>
</div>
`;
